### DISCUSSION TITLE
Repetir conteúdo em site HTML.

### DISCUSSION THREAD

**Discussion by ranosi** [frontendbr/forum#329] [Discussão]
Salve galera! Quando vocês precisam montar sites apenas em html, simples como fazem pra manter algumas informações que se repetem em todas as páginas? Tipo o topo por exemplo que é igual em todo o site.
Já vi uma vez a solução em php que poderia separar as partes e usar o include.
Não sei se ficou muito claro a questão.

**Comment by marcoaure** [comment]
html puro só ctrl+c mesmo... ou usa js ou php pra fzr esses includes

Atenciosamente, _Marco Aurélio_.  _Perfil http://marcoaure.net *
Analista-Desenvolvedor de sistemas e especialista: *Php__, *_Mysql**, *
*Html5**, **CSS3**, **JQuery**, **Front *e *Back-End**.*

Em 17 de outubro de 2016 18:48, ranosi notifications@github.com escreveu:

**Comment by thiamsantos** [comment]
@ranosi você pode usar uma template engine para fazer esses includes juntamente com alguma tarefa de build. Algumas engines boas são:
- [Nunjucks](https://mozilla.github.io/nunjucks/)
- [Pug](https://github.com/pugjs/pug)
- [EJS](http://ejs.co/)

**Comment by guilhermejcgois** [comment]
Você pode usar [HTML Templates](http://caniuse.com/#search=html%20templates), mas sem suporte para IE. Se for o caso acho que esse [artigo](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) na MDN pode te ajudar.

**Comment by ghost** [comment]
@ranosi outras template engines:

**Front-end (JavaScript):**
- [Handlebars](http://handlebarsjs.com/) (usa o Mustache)
- [Mustache](http://mustache.github.io/) (versão standalone)

**Back-end (PHP):**
- [Blade](https://github.com/PhiloNL/Laravel-Blade) (padrão do Laravel, versão standalone)
- [Scurvy](https://github.com/dburkart/scurvy) (apenas dois arquivos)
- [Smarty](http://www.smarty.net/)
- [Twig](http://twig.sensiolabs.org/) (padrão do Symfony e do Silex)

**Comment by ranosi** [comment]
Show de bola, vou dar uma olhada neles. O Twig é o único que eu tinha visto dessa lista. Valeu galera.

**Comment by giioohbernini** [comment]
@ranosi 

Tem o Jade também!! 

Mas pode ser que saia do "simples" caso você não use node.js, mas eu ando usando bastante!

Tem um artigo muito bom sobre ele [aqui](http://imasters.com.br/desenvolvimento/brincando-com-o-template-engine-jade/?trace=1519021197&source=single)

**Comment by herberthenrique** [comment]
@giioohbernini O Jade foi renomeado para Pug.
Jade era uma marca registrada e eles tiveram que renomear
https://github.com/pugjs/pug#rename-from-jade

**Comment by notnotgabriel** [comment]
Fiz um site inteiro com [Pug](https://github.com/pugjs/pug), na época ainda se chamava Jade.
Foi bastante produtivo usá-lo para um site que era somente html + css "basicão".

**Comment by Lucasktrindade** [comment]
Esses templates fazem compilação em tempo de produção, ou eles pré compilam o conteúdo que você quer e gera um HTML?

**Comment by thiamsantos** [comment]
@Lucasktrindade tanto faz. Você pode gerar o HTML estático ou integrar com o backend facilmente usando o [expressjs](http://expressjs.com/en/guide/using-template-engines.html) ou qualquer outro servidor nodejs.

**Comment by Lucasktrindade** [comment]
Saquei, no meu caso seria para gerar estático.

**Comment by gabrieldarezzo** [comment]
Em php

Eu uso bastante esse cara:
http://raelcunha.com/template/

**Comment by marinhoOliv** [comment]
Fala @ranosi, o @willianjusten tem uma curso bem legal sobre sites estáticos.
http://willianjusten.teachable.com/courses/criando-sites-estaticos-com-jekyl

**Comment by ranosi** [comment]
@marinhoOliv bem legal heim. Vou dar uma olhada nesse curso aí cara, valeu a dica.

**Comment by vitor-mendrone-deel** [comment]
Uma alternativa ao Jekyll é o [Hugo](https://gohugo.io/), que é feito em Go e por isso tem menos problemas pra rodar no Windows (Reza a lenda que Jekyll não cai muito bem no Windows por ser Ruby, porém nunca tive problemas)

**Comment by davidsonfellipe** [comment]
@ranosi Tenho usado o assemble (http://assemble.io/) que é um static site generator feito em Node.js, e tem um bom suporte para várias template engines.

**Comment by felipefialho** [comment]
Só reforçando que o Jade mudou de nome e se tornou o Pug, mas é a mesma ferramenta.

Sou fã do Pug/Jade alias, especialmente se usar Stylus, a forma de escrita fica totalmente concisa.

Caso seja mais tradicional quanto as sintaxes originais, pode usar o Nunjucks ou o EJS que o @thiamsantos citou.
